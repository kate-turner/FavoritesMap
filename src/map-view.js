import React, {useState, useEffect} from 'react';
import mapboxGl from 'mapbox-gl';
import style from './data/style.json';
import {reverseGeocode, saveToFavorites} from "./utils";
import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';


// If there are issues, replace with your token
const ACCESS_TOKEN = 'pk.eyJ1IjoiZGFzdWxpdCIsImEiOiJjaXQzYmFjYmkwdWQ5MnBwZzEzZnNub2hhIn0.EDJ-lIfX2FnKhPw3nqHcqg';

export default function MapView(props) {
    const {activeViewPort} = props

    const mapContainer = React.createRef();
    const [map, setMap] = useState(null)
    const [geoCodeCoord, setGeoCodeCoord] = useState(null)


    useEffect(() => {
        const containerEl = mapContainer;
        if (containerEl && containerEl.current) {
            mapboxGl.accessToken = ACCESS_TOKEN;
            const map = new mapboxGl.Map({
                container: containerEl.current,
                style: style,
                // center: [-122.396449, 37.791256],
                // zoom: 15
                ...activeViewPort
            });
            setMap(map);
            map.addControl(
                new MapboxGeocoder({
                    accessToken: ACCESS_TOKEN,
                    mapboxgl: mapboxGl,
                    reverseGeocode: true
                })
            );
            map.on("click", (e) => {
                console.log(e)
                setGeoCodeCoord(e.lngLat)
                reverseGeocode(e.lngLat)
                // saveToFavorites(e.lngLat)
            })
        }
    }, [])



    return (
        <div ref={mapContainer} className="map-container"/>
    )
}

export {MapView}
