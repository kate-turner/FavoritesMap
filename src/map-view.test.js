import React from 'react';
import { MapView } from './map-view';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import mapboxGl from 'mapbox-gl';

const mockInstantiationTracker = jest.fn()

jest.mock('mapbox-gl', () => ({
  Map: jest.fn()
}));

describe('MapView', () => {
  let wrapper;
  let mockFn = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
    mapboxGl.Map.mockImplementation(() => {
      return {
        mockMethod: mockFn
      };
    });
    wrapper = mount(<MapView />)
  });
  it('renders', () => {
    expect(mountToJson(wrapper)).toMatchSnapshot();
  })
  it('initializes mapbox gl', () => {
    expect(mapboxGl.Map).toHaveBeenCalledTimes(1);
  })
})