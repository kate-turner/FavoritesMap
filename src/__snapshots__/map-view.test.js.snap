// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView renders 1`] = `
<MapView>
  <div
    className="map-container"
  />
</MapView>
`;
